### PR TITLE
Fixed how dates changes are handled in different calendar event types.

### PR DESCRIPTION
**Pull request by sponkrashin**
Issue #359

### CODE REVIEW

File: server/Arcadia.Assistant.Calendar/SickLeave/EmployeeSickLeaveActor.cs [hunk 1]
```diff
@@ -176,6 +192,14 @@ protected override bool IsStatusTransitionAllowed(string oldCalendarEventStatus,
                 && (newCalendarEventStatus != SickLeaveStatuses.Approved);
         }
 
+        protected override bool IsDatesChangedAllowed(CalendarEvent oldEvent, CalendarEvent newEvent)
+        {
+            var approvals = this.ApprovalsByEvent[oldEvent.EventId];
+
+            return oldEvent.IsPending && approvals.Count == 0 ||
```
Comment (santee) [inline server/Arcadia.Assistant.Calendar/SickLeave/EmployeeSickLeaveActor.cs]: well, that needs to be discussed. Put a TODO here, I'm not sure if its right

File: server/Arcadia.Assistant.Calendar/SickLeave/EmployeeSickLeaveActor.cs [hunk 2]
```diff
@@ -188,6 +212,20 @@ private void OnSickLeaveRequest(SickLeaveIsRequested message)
             this.ApprovalsByEvent[message.EventId] = new List<Approval>();
         }
 
+        private void OnSickLeaveDatesEdit(SickLeaveDatesAreEdited message)
```
Comment (santee) [inline server/Arcadia.Assistant.Calendar/SickLeave/EmployeeSickLeaveActor.cs]: we do have Prolong. Basically that's what should be allowed. Simple dates changes - no

File: server/Arcadia.Assistant.Calendar/SickLeave/EmployeeSickLeaveActor.cs (hunk 1 above)
Comment (sponkrashin) [reply server/Arcadia.Assistant.Calendar/SickLeave/EmployeeSickLeaveActor.cs]: If later we decide to keep as is - these TODOs will stay forever, nobody will remove them :)

File: server/Arcadia.Assistant.Calendar/SickLeave/EmployeeSickLeaveActor.cs (hunk 2 above)
Comment (sponkrashin) [reply server/Arcadia.Assistant.Calendar/SickLeave/EmployeeSickLeaveActor.cs]: In fact, every employee is just a people, and he can do a mistake. Will he really have to cancel existing event and create new? It is absurdity.

Comment (sponkrashin) [reply server/Arcadia.Assistant.Calendar/SickLeave/EmployeeSickLeaveActor.cs]: This is allowed ONLY if calendar event is in requested status. So, mostly it means that event was just created.

File: server/Arcadia.Assistant.Calendar/SickLeave/EmployeeSickLeaveActor.cs (hunk 1 above)
Comment (sponkrashin) [reply server/Arcadia.Assistant.Calendar/SickLeave/EmployeeSickLeaveActor.cs]: We can remark them in separate issue as to be discussed later.

Comment (sponkrashin) [reply server/Arcadia.Assistant.Calendar/SickLeave/EmployeeSickLeaveActor.cs]: Reverted.

File: server/Arcadia.Assistant.Calendar/SickLeave/EmployeeSickLeaveActor.cs (hunk 2 above)
Comment (sponkrashin) [reply server/Arcadia.Assistant.Calendar/SickLeave/EmployeeSickLeaveActor.cs]: Reverted.

**Pull request merged**